### PR TITLE
SkillCalc optimization

### PR DESCRIPTION
**Pull request by GodsWithin**
- No longer able to select multiple checkboxes (if there are multiple)
- Made xp bonus' naming more consistant (changed some .json files)
- Made UICombinedActionSlot text more accurate (double as xp and correct amount of actions)

### CODE REVIEW

**Comment by MESLewis** [comment]
You can force push your branch to your previous PR to avoid opening a new one

**Comment by GodsWithin** [comment]
@MESLewis I tried, couldn't reopen the original PR so this will have to do.

**Comment by psikoi** [comment]
Can you add some preview images to this PR?

**Comment by psikoi** [comment]
Update the preview image while you're at it :P

**Comment by psikoi** [comment]
I'll review and test today

**Comment by GodsWithin** [comment]
Made it so that the selected actions get reset if the user changes the skill

**Comment by GodsWithin** [comment]
@psikoi Do you see any reason why I failed Travis?

**Comment by psikoi** [comment]
Hum, I can't unselect a modifier after selecting it, i can only switch to another one.

**Comment by psikoi** [comment]
I suggest keeping the jcheckboxes for single boosts, and having the radio buttons for multiple options, that's what they should be used for anyways.

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java [hunk 1]
```diff
@@ -171,28 +176,52 @@ private void renderBonusOptions()
 	{
 		if (skillData.getBonuses() != null)
 		{
+			List<JCheckBox> uiCheckBoxList = new ArrayList<>();
+			lastBonus = 0.0f;
 			for (SkillDataBonus bonus : skillData.getBonuses())
```
Comment (psikoi) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: Add line above

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java [hunk 2]
```diff
@@ -171,28 +176,52 @@ private void renderBonusOptions()
...
 				uiLabel.setFont(FontManager.getRunescapeSmallFont());
 
 				uiOption.setBorder(BorderFactory.createEmptyBorder(3, 7, 3, 0));
 				uiOption.setBackground(ColorScheme.DARKER_GRAY_COLOR);
 
-				// Adjust XP bonus depending on check-state of the boxes.
-				uiCheckbox.addActionListener(e -> adjustXPBonus(uiCheckbox.isSelected(), bonus.getValue()));
-				uiCheckbox.setBackground(ColorScheme.MEDIUM_GRAY_COLOR);
+				JCheckBox uiCheckBox = new JCheckBox();
+				uiCheckBoxList.add(uiCheckBox);
```
Comment (psikoi) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: Add the checkbox to the list after you're done messing with it, makes the code cleaner

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java [hunk 3]
```diff
@@ -171,28 +176,52 @@ private void renderBonusOptions()
...
+				uiCheckBox.setBackground(ColorScheme.MEDIUM_GRAY_COLOR);
+				uiCheckBox.addActionListener(e ->
+				{
+					if (uiCheckBox.isSelected())
+					{
+						adjustXPBonus(uiCheckBox.isSelected(), bonus.getValue());
+						lastBonus = bonus.getValue();
+						for (JCheckBox checkBox : uiCheckBoxList)
```
Comment (psikoi) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: Add line above

File: runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java [hunk 4]
```diff
@@ -171,28 +176,52 @@ private void renderBonusOptions()
...
+				});
 
+				uiOption.add(uiCheckBox, BorderLayout.EAST);
 				uiOption.add(uiLabel, BorderLayout.WEST);
-				uiOption.add(uiCheckbox, BorderLayout.EAST);
 
 				add(uiOption);
-				add(Box.createRigidArea(new Dimension(0, 5)));
 			}
+			add(Box.createRigidArea(new Dimension(0, 5)));
```
Comment (psikoi) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java]: Add line above

**Comment by GodsWithin** [comment]
If @Kruithne or @Adam- could please varify that I didn't break something, would be very much appreciated :)

**Comment by Kruithne** [comment]
Unable to test it at the moment, but I can't see anything immediately wrong with it from looking at the code.

**Comment by psikoi** [comment]
@Nightfirecat same here, #4373

**Comment by Nightfirecat** [comment]
Given the age of this PR, I will close it as stale. If you'd like to add the other changes, feel free to re-open/rebase with that done.

**Pull request closed without merging**